### PR TITLE
Avoid CommandBarFlyoutCommandBar clipping after dynamic command bar element change 

### PR DESCRIPTION
**Pull request by RBrid**
CommandBarFlyoutCommandBar did not support dynamic changes of its secondary AppBarButton/AppBarToggleButton elements.
For instance when lengthening the Label of an element after the command bar was opened, it could get clipped on the right side. The CommandBarFlyoutCommandBar did not update its UI in response to an element dependency property change.

CommandBarFlyout is now listening to AppBarButton/AppBarToggleButton dependency property changes that may affect the desired sizing of the CommandBarFlyoutCommandBar. The change notifications are forwarded to the CommandBarFlyoutCommandBar which triggers a UI refresh only when it's already open. The template settings are refreshed and the Collapsed visual state is synchronously / momentarily visited to refresh all visual aspects.

I commented out all the debug tracing that use string literals to avoid bloating the binaries.

Updated MUXControlsTestApp to allow asynchronous Label changes and asynchronous secondary commands clearing.
Tested with MUXControlsTestApp and Windows 11.
Added new interactive regression test.

### CODE REVIEW

File: dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs [hunk 1]
```diff
@@ -909,5 +910,80 @@ public void VerifyFlowsToAndFromConnectsPrimaryAndSecondaryCommandsWithAlwaysExp
...
+                UIObject commandBarElementsContainer = undoButton6.Parent;
+                Verify.IsNotNull(commandBarElementsContainer);
+
+                Rectangle initialBoundingRectangle = commandBarElementsContainer.BoundingRectangle;
+
+                Log.Comment("commandBarElementsContainer.BoundingRectangle.Width=" + initialBoundingRectangle.Width);
+                Log.Comment("commandBarElementsContainer.BoundingRectangle.Height=" + initialBoundingRectangle.Height);
+
+                Log.Comment("Waiting for the 1s before the asynchronous Label property change");
+                Wait.ForSeconds(1);
```
Comment (llongley) [inline dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs:969]: This seems both potentially unstable and kind of ugly.  Could we instead set a CheckBox in the app when the label changes and listen for its value change to detect when the label property has changed?  That's how I've typically communicated to interaction tests that a thing we were waiting for has occurred.

File: dev/CommandBarFlyout/CommandBarFlyout.cpp [hunk 2]
```diff
@@ -14,6 +14,26 @@
 bool CommandBarFlyoutTrace::s_IsDebugOutputEnabled{ false };
 bool CommandBarFlyoutTrace::s_IsVerboseDebugOutputEnabled{ false };
 
+// List of AppBarButton dependency properties being listened to for raising the CommandBarFlyoutCommandBar::OnCommandBarElementDependencyPropertyChanged notifications.
+const winrt::DependencyProperty CommandBarFlyout::s_appBarButtonDependencyProperties[s_commandBarElementDependencyPropertiesCount]
+{
+    winrt::AppBarButton::IconProperty(),
+    winrt::AppBarButton::IsCompactProperty(),
```
Comment (llongley) [inline dev/CommandBarFlyout/CommandBarFlyout.cpp:21]: Does this and LabelPosition actually affect the sizing?  I don't believe they affect AppBar*Buttons that are in the secondary commands.

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ CommandBarFlyoutCommandBar::CommandBarFlyoutCommandBar()
     Loaded({
         [this](auto const&, auto const&)
         {
-            COMMANDBARFLYOUT_TRACE_INFO(*this, TRACE_MSG_METH, METH_NAME, this);
+            //COMMANDBARFLYOUT_TRACE_INFO(*this, TRACE_MSG_METH_STR, METH_NAME, this, L"Loaded");
```
Comment (llongley) [inline dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:22]: You've left a lot of these commented out.  Is there a reason not to just delete them?

File: dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs [hunk 4]
```diff
@@ -4,6 +4,7 @@
 using Common;
 using System;
 using System.Collections.Generic;
+using System.Drawing;
```
Comment (llongley) [inline dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs:7]: What's this used for?

**Review by llongley — CHANGES_REQUESTED** [review]

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp (hunk 3 above)
Comment (RBrid) [reply dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:22]: Yes, see my comment in my opening description. I did not want to add more literals to the dll, and at the same time I want to easily re-enable them in future debugging.

File: dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs (hunk 4 above)
Comment (RBrid) [reply dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs:7]: It's used for:
. Rectangle initialBoundingRectangle
. Rectangle finalBoundingRectangle

File: dev/CommandBarFlyout/CommandBarFlyout.cpp (hunk 2 above)
Comment (RBrid) [reply dev/CommandBarFlyout/CommandBarFlyout.cpp]: Ah, my new testing showed that LabelPosition and IsCompact have no effect on AppBarButton and AppBarToggleButton as secondary commands. I took them out of the list.  Icon does have an effect and is kept. Thanks!

File: dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs (hunk 1 above)
Comment (RBrid) [reply dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs]: True. I introduced a new checkbox and wait for its signal that the Label was changed. Thanks.

**Comment by zhuxb711** [comment]
Are this related to this issue? #4878

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp (hunk 3 above)
Comment (llongley) [reply dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:22]: Ah, gotcha, so you're wanting them there so you can uncomment them if need be.  That makes sense.

**Review by llongley — APPROVED** [review]

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp (hunk 3 above)
Comment (StephenLPeters) [reply dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:22]: A Preprocessor directive is probably better than commented out code for this.

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp [hunk 5]
```diff
@@ -471,16 +475,19 @@ void CommandBarFlyoutCommandBar::UpdateFlowsFromAndFlowsTo()
 }
 
 void CommandBarFlyoutCommandBar::UpdateUI(
-    bool useTransitions)
+    bool useTransitions, bool isForCommandBarElementDependencyPropertyChange)
 {
+    COMMANDBARFLYOUT_TRACE_VERBOSE(*this, TRACE_MSG_METH_INT_INT, METH_NAME, this, useTransitions, isForCommandBarElementDependencyPropertyChange);
```
Comment (StephenLPeters) [inline dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:480]: >COMMANDBARFLYOUT_TRACE_VERBOSE [](http://example.com/codeflow?start=4&length=30)

should this be commented out as well?

**Comment by StephenLPeters** [comment]
@RBrid the KeyboardAcceleratorTextOverrideProperty DP was added to AppBarButton in RS4 which explains the down level test failures.

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp (hunk 5 above)
Comment (RBrid) [reply dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:496]: This one is fine, without literals. I added "#ifdef _DEBUG" like ScrollView::OnPropertyChanged and others do.

**Pull request merged**